core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -110,21 +110,32 @@ public void commitState(Snapshot snapshot)
       throws CommitException, UnknownTransactionStatusException {
     String id = snapshot.getId();
     try {
-      commitState(snapshot.getId());
-    } catch (CoordinatorException e) {
-      TransactionState state = abort(id);
-      if (state.equals(TransactionState.ABORTED)) {
-        rollbackRecords(snapshot);
-        throw new CommitException(
-            "committing state in coordinator failed. the transaction is aborted", e);
+      Coordinator.State state = new Coordinator.State(id, TransactionState.COMMITTED);
+      coordinator.putState(state);
+      logger.debug(
+          "transaction {} is committed successfully at {}", id, System.currentTimeMillis());
+    } catch (CoordinatorConflictException e) {
+      try {
+        Optional<Coordinator.State> s = coordinator.getState(id);
+        if (s.isPresent()) {
+          TransactionState state = s.get().getState();
+          if (state.equals(TransactionState.ABORTED)) {
+            rollbackRecords(snapshot);
+            throw new CommitException(
+                "committing state in coordinator failed. the transaction is aborted", e);
```

Comment:
TxID in the log message would be useful info for trouble shootings. I think adding transaction ID field to CommitException is also good.

Reply:
Thank you. Yes, you are right. I think we should do the same thing in other parts, so I will do that in another PR.